prober/dns.go

```diff
@@ -137,6 +137,22 @@ func ProbeDNS(ctx context.Context, target string, module config.Module, registry
 
 	client := new(dns.Client)
 	client.Net = dialProtocol
+
+	// Use configured SourceIPAddress.
+	if len(module.DNS.SourceIPAddress) > 0 {
+		if srcip := net.ParseIP(module.DNS.SourceIPAddress); srcip == nil {
+			level.Warn(logger).Log("msg", "Error parsing source ip address", "srcip", module.DNS.SourceIPAddress)
```

Comment:
This should fail the probe

Reply:
Changed to Error and failure in all probes.